### PR TITLE
BZ1259566 added important box about routers on certain hosts

### PR DESCRIPTION
**Pull request by bfallonf**
@thoraxe as discussed in #781 , all I've done here if added the important box. I'll follow up in #781, but this may be all that's required here.

### CODE REVIEW

**Comment by thoraxe** [comment]
The router can't be on an unscheduleable host -- how would the router pod get scheduled?

How about:

```
Routers need to directly attach to port 80 and 443 on all interfaces on the host. Via node selectors and scheduler configuration, routers should be restricted to hosts where port 80/443 is available and
not being consumed by another service. Dedicated router or infrastructure nodes are one way to achieve this.
```

**Comment by bfallonf** [comment]
@thoraxe Sure thing. I've made some edits. Also, I linked to the scheduler docs, but there's no solid docs on nodeselector atm. There's a BZ about that[1]. Just fyi.

Let me know if there's anything else for this, but I'll also respond to #781 now.

Thanks

[1] https://bugzilla.redhat.com/show_bug.cgi?id=1253125

**Comment by bfallonf** [comment]
...aaaaand then I go to work on a different BZ and find the ports info ( #684 ). Is this what you meant?

**Comment by thoraxe** [comment]
Yes, there are multiple issues open regarding ports and for whatever reason we still have not closed them. Those should be a very high priority.

**Comment by bfallonf** [comment]
@thoraxe You're right, there's a few open issues about not enough port-related info in the docs. I've brought it up with @vikram-redhat and we'll see if we can get something happening soon. Also, if you have anything you feel is urgent, feel free to bring it up with @CowboysFan (Bilhar is our Content Strategist who determines the urgency of docs work) or feel free to email one of us.

As for this current PR, do you think it's enough to satisfy the issue?

File: install_config/install/deploy_router.adoc [hunk 1]
```diff
@@ -99,6 +99,15 @@ communication between OpenShift components is secured by TLS and uses various
 certificates and authentication methods. Use the `--credentials` option to
 specify what credentials the router should use to contact the master.
 
+[IMPORTANT]
+====
+Attach routers directly to port 80 and 443 on all interfaces on a host. Restrict
```
Comment (thoraxe) [inline install_config/install/deploy_router.adoc]: The user doesn't attach the routers -- the routers attach themselves.

Perhaps:

"Routers directly attach to port 80 and 443 on all interfaces on the hosts they run on."

?

**Comment by bfallonf** [comment]
@thoraxe Except for the "...they run on" (because preposition), suggestion fixed. If there's nothing else here, this can probably be ready to merge. I'll make sure to follow up on the ports doc.

File: install_config/install/deploy_router.adoc [hunk 2]
```diff
@@ -99,6 +99,15 @@ communication between OpenShift components is secured by TLS and uses various
 certificates and authentication methods. Use the `--credentials` option to
 specify what credentials the router should use to contact the master.
 
+[IMPORTANT]
+====
+Routers directly attach to port 80 and 443 on all interfaces on a host. Restrict
+routers to hosts where port 80/443 is available and not being consumed by
+another service, and set this using node selectors and the
+link:admin_guide/scheduler.html[scheduler configuration]. As an example, you can
+achieve this with a dedicated router or infrastructure nodes.
```
Comment (adellape) [inline install_config/install/deploy_router.adoc]: I think it should be "with dedicated router or infrastructure nodes", not "a dedicated router". @thoraxe ack?

**Comment by adellape** [comment]
LGTM other than the one inline comment.

File: install_config/install/deploy_router.adoc (hunk 2 above)
Comment (thoraxe) [inline install_config/install/deploy_router.adoc]: "As an example, you can achieve this with dedicated "infrastructure" nodes where your routers will run" or something like that.

Comment (adellape) [inline install_config/install/deploy_router.adoc]: +1

Comment (bfallonf) [inline install_config/install/deploy_router.adoc]: @adellape @thoraxe How about:

"As an example, you can achieve this with infrastructure nodes, which are dedicated to run services such as routers."

Comment (thoraxe) [inline install_config/install/deploy_router.adoc]: "As an example, you can achieve this by dedicating infrastructure nodes to run services such as routers."

Active tense! :)

Comment (adellape) [inline install_config/install/deploy_router.adoc]: > Active tense! :)

+1

Comment (adellape) [inline install_config/install/deploy_router.adoc]: Side thought: In general, we don't ever actually discuss the `infra` region or its intended purpose in default installs. Right now you'd just have to notice the labels in the advanced install inventory files and assume. Should probably mention them more explicitly in the install sections (and/or Managing Nodes topic?).

For another PR tho (wouldn't be surprised if it's mentioned in an outstanding Issue, tho a quick search doesn't hit anything).

Comment (thoraxe) [inline install_config/install/deploy_router.adoc]: We could use a section on "topology" in relation to the scheduler which is where we could introduce the topic... but that's more part of the architecture overview section I'd think.

Comment (adellape) [inline install_config/install/deploy_router.adoc]: More +1s

**Comment by bfallonf** [comment]
Edited to Erik's suggestion. I'm presuming there's nothing else, so I'll merge this.

I'll also make a BZ about mentioning infrastructure nodes.

**Comment by adellape** [comment]
@bfallonf Added "OSE 3.0.z" milestone so we'll know it can get picked up in the enterprise-3.0 branch.

**Pull request merged**